src/components/TransactionsTab/TransactionsTab.tsx

```diff
@@ -150,6 +153,14 @@ const TransactionsTab = ({
 
   const balance = balances.values[nativeCurrencyAddress] || "0";
 
+  useEffect(() => {
+    if (completedTransactions.length > 1 && pendingTransactions.length === 0) {
+      setTransactionComplete(true);
+    } else {
+      setTransactionComplete(false);
+    }
+  }, [completedTransactions, pendingTransactions, setTransactionComplete]);
+
```

Comment:
You should write this kind of logic in the transactions store and not in a component. You should also look for a specific transaction, you should have a nonce when you submit a transaction, just wait for it to complete and your transaction is complete.

Reply:
I think you should just make a hook, look at `useCancellationSuccess` for inspiration.